### PR TITLE
ieee802154_hal: add cap for register retention during `off`

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR adds a radio cap for indicating that the radio can keep the content of all register when off. This is required to indicate the upper layer that in some cases it might be necessary to set the PIB again.

This still leaves room for several use cases:
- Users that don't require register retention (because the CPU hibernates) and simply reboot on wake-up can ignore setting the PIB after `off`.
- Users that require register retention (radio sleeps but CPU is still awake) can set the PIB if the cap is not there.

It also specifies that during boot, the PIB is undefined (so it should be set once `confirm_on` returns 0).

I also piggybacked a misplaced note that was probably a left-over of the cleanup in #14371 

### Testing procedure
The output of Travis should be enough

### Issues/PRs references
Discussed in #14802

### CODE REVIEW

File: sys/include/net/ieee802154/radio.h [hunk 1]
```diff
@@ -484,6 +488,18 @@ struct ieee802154_radio_ops {
      * @pre call to @ref ieee802154_radio_ops::request_on was successful.
      *
      * @post the transceiver state is @ref IEEE802154_TRX_STATE_TRX_OFF
+     * During boot or in case the radio doesn't support @ref
+     * IEEE802154_CAP_REG_RETENTION when @ref off was called, the
+     * Physical Information Base will be undefined. Thus, for take into
```
Comment (benpicco) [inline sys/include/net/ieee802154/radio.h:493]:
```suggestion
     * Physical Information Base will be undefined. Thus, take into
```

Either there is a word missing or one too much ;)

Comment (jia200x) [reply sys/include/net/ieee802154/radio.h]: done!

Comment (benpicco) [reply sys/include/net/ieee802154/radio.h]: Please squash!

**Review by benpicco — APPROVED** [review]
Good addition, only API defined so far.

**Comment by jia200x** [comment]
squashed

**Pull request merged**